SIGSEGV in span_writer

I hit this when deploying `quay.io/signalfx/signalfx-agent:5.3.3` via Helm using chart version 1.0.1. 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17bba55]

goroutine 45 [running]:
github.com/signalfx/signalfx-agent/pkg/core/writer/tracetracker.(*SpanSourceTracker).AddSourceTagsToSpan(0xc000ecb8c0, 0xc0017230e0)
	/usr/src/signalfx-agent/pkg/core/writer/tracetracker/host.go:60 +0x335
github.com/signalfx/signalfx-agent/pkg/core/writer/signalfx.(*Writer).processSpan(0xc000eee120, 0xc0017230e0, 0xc00043cea0)
	/usr/src/signalfx-agent/pkg/core/writer/signalfx/spans.go:39 +0x66
github.com/signalfx/signalfx-go/writer.(*SpanWriter).processInput(0xc00039c320, 0x2c36240, 0xc000ec2a00, 0xc00019c978, 0x1, 0x1)
	/go/pkg/mod/github.com/signalfx/signalfx-go@v1.6.38-0.20200518153434-ceee8d2570d5/writer/span_writer.gen.go:177 +0x208
github.com/signalfx/signalfx-go/writer.(*SpanWriter).run(0xc00039c320, 0x2c36240, 0xc000ec2a00)
	/go/pkg/mod/github.com/signalfx/signalfx-go@v1.6.38-0.20200518153434-ceee8d2570d5/writer/span_writer.gen.go:303 +0x5f4
github.com/signalfx/signalfx-go/writer.(*SpanWriter).Start.func1(0xc00039c320, 0x2c36240, 0xc000ec2a00)
	/go/pkg/mod/github.com/signalfx/signalfx-go@v1.6.38-0.20200518153434-ceee8d2570d5/writer/span_writer.gen.go:212 +0x3f
created by github.com/signalfx/signalfx-go/writer.(*SpanWriter).Start
	/go/pkg/mod/github.com/signalfx/signalfx-go@v1.6.38-0.20200518153434-ceee8d2570d5/writer/span_writer.gen.go:211 +0x84
```

I did notice the helm chart did update, not sure if it's a problem in the image or just some incompatibility with chart/app version?

--- a/pkg/core/writer/tracetracker/host.go
+++ b/pkg/core/writer/tracetracker/host.go
@@ -57,7 +57,7 @@ func (st *SpanSourceTracker) AddSourceTagsToSpan(span *trace.Span) {
 			if val := dims[dim]; val != "" {
 				found++
 
-				if span.LocalEndpoint.ServiceName != nil {
+				if span.LocalEndpoint != nil && span.LocalEndpoint.ServiceName != nil {
 					st.emitDimensionPropIfNew(dim, val, *span.LocalEndpoint.ServiceName)
 				}
 

